src/app/public/src/modules/open-on-click-directive/open-on-click.directive.ts

```diff
@@ -0,0 +1,40 @@
+import { Directive, HostListener, Input, Renderer2, ElementRef } from '@angular/core';
+
+import { HelpWidgetService } from '../shared';
+
+@Directive({
+  selector: '[bbHelpOpenOnClick]'
+})
+export class BBHelpOpenOnClickDirective {
+  @Input()
+  public bbHelpOpenOnClick: string;
+
+  constructor (
+    private widgetService: HelpWidgetService,
+    private elementRef: ElementRef,
+    private renderer: Renderer2
+  ) {
+    this.renderer.setStyle(this.elementRef.nativeElement, 'cursor', 'pointer');
+  }
+
+  @HostListener('click', ['$event'])
+  @HostListener('keydown', ['$event'])
+  public onFocusEvent(event: MouseEvent | KeyboardEvent): boolean {
```

Comment:
Might be easier to parse which event does what if you split them up:

```
@HostListener('click', ['$event'])
public onClick(event: any) {
  this.openWidget();
  event.preventDefault();
}

@HostListener('keydown', ['$event'])
public onKeydown(event: any) {
  if (event.key.toLocaleLowerCase() === 'enter') {
    this.onClick(event);
  }
}
```

Reply:
I went back and forth on this, it seemed 6 of one half dozen of the other.  Though it does seem cleaner to have them split, just wasn't fully sold on needing a separate method for an `if` statement.

Reply:
The fact that you were checking if the event was an instance of `KeyboardEvent` is what got me; wouldn't need that check if they were separated.

Reply:
👍 Fair! and updated